pkg/analyzer/lib/src/generated/engine.dart

```diff
@@ -840,6 +841,7 @@ class AnalysisEngine {
    */
   void clearCaches() {
     partitionManager.clearCache();
+    StringToken.canonicalizer.clear();
```

Comment:
I searched for other uses of this method and found a comment in `pkg/compiler/lib/src/compiler.dart`:
```
    // TODO(ahe): This prevents memory leaks when invoking the compiler
    // multiple times. Implement a better mechanism where we can store
    // such caches in the compiler and get access to them through a
    // suitably maintained static reference to the current compiler.
    StringToken.canonicalizer.clear();
```
Please add a comment here referencing https://github.com/dart-lang/sdk/issues/30314 so that we know the symptom if/when we change fasta string canonicalization.

Thanks!